### ISSUE TITLE
Consider working around T_HASHBANG in HHVM 3.5.x and 3.6.x

### ISSUE THREAD

**Issue by legoktm** [squizlabs/PHP_CodeSniffer#625]
Some versions of HHVM's `token_get_all()` function will parse a hashbang like `#!/usr/bin/php` as T_HASHBANG instead of the T_INLINE_HTML that PHP5 uses. 

This has been fixed in HHVM 3.7 (https://github.com/facebook/hhvm/pull/5023), but does it make sense to add a workaround for this in PHPCS? https://gerrit.wikimedia.org/r/219394 is my fix for MediaWiki's custom sniff.

3.6 (an affected version) is an HHVM LTS that will be supported until January 28, 2016.

**Comment by gsherwood** [comment]
Just confirming this behaviour.

This code
```php
#!/usr/bin/php
<?php
echo "foo";
```
tokenizes as
```
0 : T_INLINE_HTML => #!/usr/bin/php\n
1 : T_OPEN_TAG => <?php\n
2 : T_ECHO => echo
3 : T_WHITESPACE =>  
4 : T_CONSTANT_ENCAPSED_STRING => "foo"
5 : ;
6 : T_WHITESPACE => \n
```
except in HHVM 3.6, where it is tokenized like this
```
0 : T_HASHBANG => #!/usr/bin/php\n
1 : T_OPEN_TAG => <?php\n
2 : T_ECHO => echo
3 : T_WHITESPACE =>  
4 : T_CONSTANT_ENCAPSED_STRING => "foo"
5 : ;
6 : T_WHITESPACE => \n
```
As there is no T_HASHBANG token in PHP itself, converting this over to T_INLINE_HTML should be pretty easy and shouldn't cause any errors.
